DetectorDescription/DDCMS/src/DDDetector.cc

```diff
@@ -11,7 +11,7 @@
 
 namespace cms {
 
-  DDDetector::DDDetector(const std::string& tag, const std::string& fileName, bool bigXML) : m_tag(tag) {
+  DDDetector::DDDetector(const std::string& tag, const std::string& fileName, bool bigXML, bool validate) : m_tag(tag) {
```

Comment:
The `validate` parameter is not needed. The `bigXML` parameter says whether the geometry is coming from the DB. If the geometry is from the DB (`bigXML == true`), like in production, we can assume the geometry has been checked, so the validate checks are not need.
If the geometry comes from the XML files (`bigXML == false`), the job is not a production job and is probably being run during development or by a PR test. In this case, we want the validation, so the `validate_` member can be set based upon the `bigXML` parameter.

Reply:
If that is what you want, we can do that. However, since the work was already done, wouldn't it be useful to also to be able to double check any DB payloads?

Reply:
DB payloads are generated from the XML files. We can always check the source XML files. Also, we have test scripts that compare DB payload contents with the XML geometry to make sure that they match. There really isn't a use-case for independent validation of the DB payloads, and the `validate` parameter just adds complexity.
By using the `bigXML` parameter to determine validation, we avoid having to propagate the validation parameter throughout a lot of Python configs, and we guarantee that production jobs running from the DB will not do these unnecessary validation checks.